### PR TITLE
Deploy OCI loadbalancer for Istio

### LINKED ISSUE #12: Create second load balancer for Istio
Deploy a second OCI Loadbalancer that points to the NodePorts that the Istio module of OLCNE uses.

### PR DESCRIPTION
**Pull request by mark-au**
Deploy a second load balancer that points at Istio NodePorts 
Outputs IP addresses of both loadbalancers, this could be extended to a new feature where DNS records are automatically modified.

Resolves #12

### CODE REVIEW

**Comment by mark-au** [comment]
There is a healthcheck for Istio but it's not on a stable port (15020 /healthz/ready) so leaving as TCP healthcheck on the HTTP port. Note that istio-ingressgateway will not start responding to queries until a Gateway object is defined in Kubernetes

**Review by Djelibeybi — DISMISSED** [review]

File: modules/loadbalancer/outputs.tf [hunk 1]
```diff
@@ -8,3 +8,11 @@ output "pub_lb_id" {
 output "pub_lb_ip" {
```
Comment (hyder) [inline modules/loadbalancer/outputs.tf]: Since you are renaming all the public load balancer to nginx, this should be renamed to nginx_lb_ip too

**Comment by hyder** [comment]
Are you also deploying the istio module as part of this or you are just preparing the networking?

**Comment by mark-au** [comment]
> Are you also deploying the istio module as part of this or you are just preparing the networking?

Just creating the OCI load balancer to point to the correct NodePorts

File: outputs.tf [hunk 2]
```diff
@@ -15,3 +15,7 @@ output "ssh_to_master" {
   description = "ssh to primary master node"
   value       = "ssh -i ${var.ssh_private_key_path} -J opc@${module.base.bastion_public_ip} opc@${module.master.master_vip}"
 }
+
+output "Loadbalancer_addresses" {
```
Comment (hyder) [inline outputs.tf]: snake_case please

**Comment by hyder** [comment]
> > Are you also deploying the istio module as part of this or you are just preparing the networking?
> 
> Just creating the OCI load balancer to point to the correct NodePorts

In that case, we should also add a variable that controls its creation i.e. it will only be created if Istio is installed

**Comment by mark-au** [comment]
> In that case, we should also add a variable that controls its creation i.e. it will only be created if Istio is installed

Any suggestions on how it should be done? I can't do something like `count = var.deploy_istio_loadbalancer ? 1 : 0` to `istio_lb` as it has non-instance based references.

**Comment by scoter-oracle** [comment]
This update is now too old compared to the updated code we bring from internal OCNE terraform repository.

**Pull request closed without merging**